### PR TITLE
:memo: Added latest paper references to the documentation

### PR DESCRIPTION
**Pull request by marcelwa**
## Description

This PR adds several of our latest paper references to the documentation.

## Checklist:

- [x] The pull request only contains commits that are related to it.
- [x] I have added appropriate tests and documentation.
- [x] I have made sure that all CI jobs on GitHub pass.
- [x] The pull request introduces no new warnings and follows the project's style guidelines.

### CODE REVIEW

File: docs/publications.rst [hunk 1]
```diff
@@ -93,43 +93,77 @@ Furthermore, if you use any of the design automation algorithms, please consider
...
           year={2023}
         }
 
+
+* :ref:`Post-layout optimization <post_layout_optimization>` (``post_layout_optimization``):
+
+    .. code-block:: tex
+
+        @inproceedings{hofmann2023postlayout,
+          title={{Post-Layout Optimization for Field-Coupled Nanotechnologies}},
```
Comment (simon1hofmann) [inline docs/publications.rst:153]:
```suggestion
          title={{Post-Layout Optimization for Field-coupled Nanotechnologies}},
```

Comment (simon1hofmann) [reply docs/publications.rst:153]: We changed the title to "Field-coupled" to align with other publications.

Comment (simon1hofmann) [reply docs/publications.rst:153]: Has been changed on the CDA website already: https://github.com/cda-tum/webpage/pull/467

**Review by Drewniok — APPROVED** [review]
LGTM, thank you!

**Pull request merged**